client/tx/factory.go

```diff
@@ -52,13 +52,16 @@ func NewFactoryCLI(clientCtx client.Context, flagSet *pflag.FlagSet) Factory {
 	gasAdj, _ := flagSet.GetFloat64(flags.FlagGasAdjustment)
 	memo, _ := flagSet.GetString(flags.FlagMemo)
 
+	gasStr, _ := flagSet.GetString(flags.FlagGas)
+	gasSetting, _ := flags.ParseGasSetting(gasStr)
```

Comment:
```suggestion
	gasSetting, err := flags.ParseGasSetting(gasStr)
	if err != nil {
		panic(err)
	}
```

This is because `ParseGasSetting` must never fail there, but if it does it must panic

Reply:
Ideally, we return an error here.